api/container_registries_proxy_scanner.go

```diff
@@ -0,0 +1,75 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+// GetProxyScanner gets a single ProxyScanner integration matching the
+// provided integration guid
+func (svc *ContainerRegistriesService) GetProxyScanner(guid string) (
+	response ProxyScannerIntegrationResponse,
+	err error,
+) {
+	err = svc.get(guid, &response)
+	return
+}
+
+// UpdateProxyScanner updates a single ProxyScanner integration on the Lacework Server
+func (svc *ContainerRegistriesService) UpdateProxyScanner(data ContainerRegistry) (
+	response ProxyScannerIntegrationResponse,
+	err error,
+) {
+	err = svc.update(data.ID(), data, &response)
+	return
+}
+
+type ProxyScannerIntegrationResponse struct {
+	Data        ProxyScannerIntegration  `json:"data"`
+	ServerToken *ProxyScannerServerToken `json:"serverToken"`
+}
+
+type ProxyScannerIntegration struct {
+	v2CommonIntegrationData
+	Data        ProxyScannerData         `json:"data"`
+	ServerToken *ProxyScannerServerToken `json:"serverToken"`
+}
+
+type ProxyScannerServerToken struct {
```

Comment:
I wonder if we can combine these three structs (inline, proxy, v2) and just have the one you define in `V2ServerToken` instead.

Reply:
done